source/vibe/http/common.d

```diff
@@ -469,20 +488,25 @@ final class ChunkedOutputStream : OutputStream {
 	{
 		assert(!m_finalized);
 		if( m_buffer.data.length > 0 ) flush();
+		ubyte[] buffer = null;
+		auto getBytes = (size_t size) {
+			if (buffer is null || buffer.length < size)
+				buffer = new ubyte[](size);
+				return buffer[0..size];
+			};
+
 		if( nbytes == 0 ){
 			while( !data.empty ){
 				auto sz = data.leastSize;
 				assert(sz > 0);
-				writeChunkSize(sz);
-				m_out.write(data, sz);
-				m_out.write("\r\n");
-				m_out.flush();
+				auto bytes = getBytes(sz);
+				data.read(bytes);
+				write(bytes);
```

Comment:
This change will result in the stream always using chunks of the size `maxBufferSize`, except for the last one. This is an implicit requirement for the S3 usecase (since chunks are not to allowed to be under a certain size).
Using the new extension mechanism it is also the easiest solution, since all data written to the stream has to be put into a buffer in this class anyway (in order to be able to call the delegate). It will however differ from the current chunking behavior. Is this a problem?